components/datetime/src/provider/neo.rs

```diff
@@ -0,0 +1,155 @@
+#[derive(Debug, PartialEq, Clone)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_datetime::provider::neo),
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[yoke(prove_covariance_manually)]
+pub struct WeekdaySymbolsV1<'data> {
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    symbols: VarZeroVec<'data, str>,
+}
+
+/// Symbols used for representing the day period
+///
+/// This uses an auxiliary key for length.
+
+#[icu_provider::data_struct(marker(DayPeriodSymbolsV1Marker, "datetime/symbols/dayperiods@1"))]
+#[derive(Debug, PartialEq, Clone)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_datetime::provider::neo),
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[yoke(prove_covariance_manually)]
+pub struct DayPeriodSymbolsV1<'data> {
+    /// Stored in the order
+    /// am, pm, (noon), (midnight)
+    /// where parentheses are optional
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    symbols: VarZeroVec<'data, str>,
+}
+
+/// Symbols used for representing the day name for calendars with named days (Chinese, Korean, Hindu)
+///
+/// This uses an auxiliary key for length.
+#[icu_provider::data_struct(marker(DaySymbolsV1Marker, "datetime/symbols/days@1"))]
+#[derive(Debug, PartialEq, Clone)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_datetime::provider::neo),
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[yoke(prove_covariance_manually)]
+pub struct DaySymbolsV1<'data> {
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    symbols: VarZeroVec<'data, str>,
+}
+
+/// The default per-length patterns associated with dates and times
+///
+/// This uses an auxiliary key for length.
+#[icu_provider::data_struct(
+    marker(DateLengthsV1Marker, "datetime/lengths/date@1"),
+    marker(TimeLengthsV1Marker, "datetime/lengths/time@1")
+)]
+#[derive(Debug, PartialEq, Clone)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_datetime::provider::neo),
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[yoke(prove_covariance_manually)]
+pub struct LengthsV1<'data> {
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pattern: runtime::Pattern<'data>,
+}
+
+/// The default hour cycle intended to be used with a locale
+#[icu_provider::data_struct(marker(PreferredHourCycleV1Marker, "datetime/lengths/hourcycle@1"))]
+#[derive(Debug, PartialEq, Clone, Copy)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_datetime::provider::neo),
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[yoke(prove_covariance_manually)]
+pub struct PreferredHourCycleV1 {
+    cycle: pattern::CoarseHourCycle,
```

Comment:
bit worried it's wasteful to have an entire key for this one tiny nugget of data.

Reply:
Not opposed. I can leave this out for now then.

Reply:
I prefer having tiny data structs and leveraging the provider infrastructure. A singleton key cannot be locale-filtered, and the filtering is actually impactful because the locales are big(ger).